msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OnBehalfOfRequest.java

```diff
@@ -30,20 +29,21 @@ class OnBehalfOfRequest extends MsalRequest {
     }
 
     private static OAuthAuthorizationGrant createAuthenticationGrant(OnBehalfOfParameters parameters) {
+        Map<String, List<String>> params = new LinkedHashMap<>();
 
-        AuthorizationGrant jWTBearerGrant;
-        try {
-            jWTBearerGrant = new JWTBearerGrant(SignedJWT.parse(parameters.userAssertion().getAssertion()));
-        } catch (Exception e) {
-            throw new MsalClientException(e);
-        }
-
-        Map<String, List<String>> params = new HashMap<>();
+        params.put(GrantConstants.GRANT_TYPE_PARAMETER, Collections.singletonList(GrantConstants.JWT_BEARER));
         params.put("requested_token_use", Collections.singletonList("on_behalf_of"));
+
         if (parameters.claims() != null) {
             params.put("claims", Collections.singletonList(parameters.claims().formatAsJSONString()));
         }
 
-        return new OAuthAuthorizationGrant(jWTBearerGrant, String.join(SCOPES_DELIMITER, parameters.scopes()), params);
+        try {
+            params.put(GrantConstants.ASSERTION_PARAMETER, Collections.singletonList(SignedJWT.parse(parameters.userAssertion().getAssertion()).getParsedString()));
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
+
+        return new OAuthAuthorizationGrant(params, String.join(SCOPES_DELIMITER, parameters.scopes()), null);
```

Comment:
I thought that the second parameter is a List<string>. but here you add a string (which becomes a list of char?)

Reply:
In the latest commit I reduced the constructors in `OAuthAuthorizationGrant` to just one, and refactored the code in several classes so that they aren't swapping back and forth between sets, lists, strings, etc.